scope.go

```diff
@@ -508,18 +500,12 @@ func (s *scope) Snapshot() Snapshot {
 }
 
 func (s *scope) Close() error {
-	s.status.Lock()
```

Comment:
nice, I've always thought the myriad attempts to make close not racy led to this huge mess.